cosmovisor/README.md

```diff
@@ -66,6 +68,18 @@ In order to support downloadable binaries, a tarball for each upgrade binary wil
 
 The `DAEMON` specific code and operations (e.g. tendermint config, the application db, syncing blocks, etc.) all work as expected. The application binaries' directives such as command-line flags and environment variables also work as expected.
 
+
+### Detecting Upgrades
+
+`cosmovisor` is polling the `$DAEMON_HOME/data/upgrade-info.json` file for new upgrade instructions. The file is created by the app x/upgrade module in the `BeginBlocker` when an upgrade is detected and the blockchain reached the upgrade height.
+The following heuristic is applied to detect the upgrade:
++ When starting, `cosmovisor` doesn't know much about currently running upgrade, except the binary which is `current/bin/` (when it the `current` directory doesn't exists then it is created by linking to `genesis`). It tries to read the `current/update-info.json` file to get information about the current upgrade name.
```

Comment:
```suggestion
+ When starting, `cosmovisor` tries to read the `current/update-info.json` file to get information about the current upgrade name.
```

Reply:
why do you want to remove the sentence?

Reply:
It's ok to leave as is, just a suggestion. How the current directory works was explained already a few paragraphs above, so it seemed a bit verbose. It is also a bit confusing referencing "the current upgrade" and then immediately after mentioning the genesis binary (not an upgrade). The essential information is how cosmovisor gets the upgrade name.

Reply:
I've updated this bullet point.